src/main/kotlin/com/example/serverFactoryAPI/security/jwt/AuthenticationTokenFilter.kt

```
@@ -0,0 +1,56 @@
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.util.StringUtils
+import org.springframework.web.filter.OncePerRequestFilter
+import java.io.IOException
+import javax.servlet.FilterChain
+import javax.servlet.ServletException
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+
+class AuthenticationTokenFilter : OncePerRequestFilter() {
+    @Autowired
+    private lateinit var jwtUtils: JwtUtils
+
+    @Autowired
+    private lateinit var userDetailsService: UserDetailsServiceImplementation
+
+    @Throws(ServletException::class, IOException::class)
+    override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
+        try {
+            println("doFilterInternal")
+            val jwt = parseJwt(request)
+            if (jwt != null && jwtUtils.validateJwtToken(jwt)) {
+                val username = jwtUtils.getUserNameFromJwtToken(jwt)
+                val userDetails: UserDetails = userDetailsService.loadUserByUsername(username)
+                val authentication = UsernamePasswordAuthenticationToken(
+                        userDetails, null, userDetails.authorities)
+                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
+                SecurityContextHolder.getContext().authentication = authentication
+            }
+        } catch (e: Exception) {
+            Companion.logger.error("Cannot set user authentication: {}", e)
+        }
+        filterChain.doFilter(request, response)
+    }
+
+    private fun parseJwt(request: HttpServletRequest): String? {
+        println("parseJwt")
+        val headerAuth = request.getHeader("Authorization")
+        return if (StringUtils.hasText(headerAuth) && headerAuth.startsWith("Bearer ")) {
```

@MarkoLazovicCode Does it make sense to throw proper exception if condition is not fulfilled instead of returning null?